### ISSUE TITLE
Question about "KDE Loss"

### ISSUE THREAD

**Issue by dreamyou070** [TencentARC/Mix-of-Show#4]
Hi,
I am interested in your paper and the code, and studying with this now,

When generate ED Lora, you used two kinds of loss, VLB loss for stable diffusion LORA layer training, and KDE Loss for new token embedding. I am wondering what KDE loss for. In the code, you compare original text embedding token embedding and the new token text encoder embedding (nn.Embedding)

**Comment by guyuchao** [comment]
Sorry to late reply. KDE loss refers to "kernel-based density estimation". It is introduced in "P+: Extended Textual Conditioning in Text-to-Image Generation" Sec 4.4. This loss imposes constraints on the new embeddings, ensuring they lie within the language manifold. However, it is not a crucial factor in ED-LoRA training, and we plan to remove it from the main branch due to its excessive memory consumption.
